oonib/oonibackend.py

```diff
@@ -151,27 +153,21 @@ def createService(endpoint, role, endpoint_config):
     multiService.addService(service)
     service.startService()
 
-torconfig = None
-if config.main.tor_hidden_service:
-    torconfig = TorConfig()
-    configTor(torconfig)
-
 # this is to ensure same behaviour with an old config file
-if config.main.bouncer_endpoints is None and config.main.tor_hidden_service:
-    config.main.bouncer_endpoints = [ {'type': 'onion', 'hsdir': 'bouncer'} ]
-
-if config.main.collector_endpoints is None and config.main.tor_hidden_service:
-    config.main.collector_endpoints = [ {'type': 'onion', 'hsdir': 'collector'} ]
-
-config.main.bouncer_endpoints = config.main.get('bouncer_endpoints', [])
-config.main.collector_endpoints = config.main.get('collector_endpoints', [])
-
-for endpoint_config in config.main.bouncer_endpoints:
+if config.main.tor_hidden_service and \
+        config.main.bouncer_endpoints is None and \
+        config.main.collector_endpoints is None:
+    bouncer_hsdir   = os.path.join(config.main.tor_datadir, 'bouncer')
+    collector_hsdir = os.path.join(config.main.tor_datadir, 'collector')
+    config.main.bouncer_endpoints   = [ {'type': 'onion', 'hsdir':   bouncer_hsdir} ]
+    config.main.collector_endpoints = [ {'type': 'onion', 'hsdir': collector_hsdir} ]
+
+for endpoint_config in config.main.get('bouncer_endpoints'):
```

Comment:
You need to pass a default to `get()` (`get('bouncer_endpoints', [])`) otherwise you will attempt to iterate over a None when bouncer_endpoints is not defined.